src/WPGraphQL.php

```diff
@@ -640,33 +622,45 @@ public static function get_allowed_post_types( $output = 'names', $args = [] ) {
 			$allowed_post_type_names = apply_filters( 'graphql_post_entities_allowed_post_types', $post_type_names, $post_type_objects );
 
 			// Filter the post type objects if the list of allowed types have changed.
-			if ( $post_type_names !== $allowed_post_type_names ) {
-				// Maybe they're just out of order.
-				sort( $post_type_names );
-				sort( $allowed_post_type_names );
-
-				if ( $post_type_names !== $allowed_post_type_names ) {
-					$post_type_objects = array_filter($post_type_objects, function ( $obj ) use ( $allowed_post_type_names ) {
-						return in_array( $obj->name, $allowed_post_type_names, true );
-					});
+			$post_type_objects = array_filter( $post_type_objects, static function ( $obj ) use ( $allowed_post_type_names ) {
+
+				if ( empty( $obj->graphql_plural_name ) && ! empty( $obj->graphql_single_name ) ) {
+					$obj->graphql_plural_name = $obj->graphql_single_name;
 				}
-			}
+
+				/**
+				 * Validate that the post_types have a graphql_single_name and graphql_plural_name
+				 */
+				if ( empty( $obj->graphql_single_name ) || empty( $obj->graphql_plural_name ) ) {
+					graphql_debug(
```

Comment:
We used to throw a UserError. Is this change to graphql_debug() intentional, and are there downstream considerations we need to worry about?

Reply:
It is intentional. 

With so many non-code ways to add post types (CPT UI, ACF (upcoming), Toolset Types, etc) it feels harsh to blow up the entire schema with an error for a missing key. Instead, we simply exclude the post type because it's missing information and can inform the user of the mistake via graphql_debug() while still allowing the existing queries, etc to continue working.